### PR TITLE
Shoot control

### PR DESCRIPTION
**Pull request by Alice-Xiong**
**Whats new:**
user/APP/CAN_receive - refactor motor names (from rev and shoot to hopper and trigger)
user/TASK/startTask - opened new task called shootTask, defined task delay times 
**user/TASK/shootTask - controlling flywheels, trigger motor, and hopper motor controlled by RC**
user/TASK/gimbalTask - integrating shootTask into gimbalTask

**What bugs were (potentially) introduced:**
The PID constants are not tuned and the trigger motor functionality is not tested, since the current infantry does not have a trigger motor. Further testing will be done once mech prototype is finished.

**Documentation Links:**
Same doc as the gimbal documentation: https://slite.com/api/s/note/HRcdULtHZHgsFrUUdA3UBy/Gimbal-and-Launcher

### CODE REVIEW

**Comment by JoshMarangoni** [comment]
We should add the following to the .gitignore

*.vscode/

File: user/APP/CAN_receive/CAN_receive.c [hunk 1]
```diff
@@ -171,31 +171,31 @@ void CAN_CMD_CHASSIS(int16_t motor1, int16_t motor2, int16_t motor3, int16_t mot
 /******************* Public CAN Read Functions *******************/
 
 //Return a pointer to yaw motor data
-const motor_measure_t *get_Yaw_Gimbal_Motor_Measure_Point(void)
+const motor_data_t *get_Yaw_Gimbal_Motor_Measure_Point(void)
```
Comment (JoshMarangoni) [inline user/APP/CAN_receive/CAN_receive.c]: This might break things

Comment (JoshMarangoni) [reply user/APP/CAN_receive/CAN_receive.c]: If we are changing the name of this structure, this needs to be reflected in gimbal_task.h in the struct Gimbal_Motor_t

Comment (jingleton) [reply user/APP/CAN_receive/CAN_receive.c]: All references to the type were/should have been updated (gimbal_task.h definitely was)

**Comment by Alice-Xiong** [comment]
@jingleton @JoshMarangoni Thanks for the comment. Yes I agree that we should not merge this until further tested. Though the purpose of having the PR up is so that any general compatibility/algorithm changes can be addressed, as they are :>

On the vscode directory: I have gitignored it for now. But if anyone does use vscode, the settings file contains a line to make it easier to tab in vscode I think. (@Xenans )

File: user/APP/CAN_receive/CAN_receive.h [hunk 2]
```diff
@@ -39,8 +39,8 @@ typedef enum
 
     CAN_YAW_MOTOR_ID = 0x205,
     CAN_PIT_MOTOR_ID = 0x206,
-    CAN_SHOOT_MOTOR_ID = 0x207,
-	CAN_REVOLVER_MOTOR_ID = 0x208,
+    CAN_TRIGGER_MOTOR_ID = 0x207,
+	CAN_HOPPER_MOTOR_ID = 0x208,
```
Comment (jamesrosstwo) [inline user/APP/CAN_receive/CAN_receive.h]: tab indent here, should be 4 spaces

File: user/TASK/shoot_task/shoot_task.c [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+
+/**
+ * @brief Trigger motor moves 90 degrees to allow one ball into the launcher
+ *        Hopper motor continues to spin to provide constant feed
+ * @param Trigger motor and hopper motor structs
+ * @retval None
+ */
+static void shoot_single_control(Shoot_Motor_t *trigger_motor, Shoot_Motor_t *hopper_motor)
+{
+    //Set target to be 1/4pi from now
```
Comment (jamesrosstwo) [inline user/TASK/shoot_task/shoot_task.c]: image

File: user/TASK/shoot_task/shoot_task.c [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+    if (trigger_motor->move_flag == 0 && shoot.mode == SHOOT_SINGLE)
+    {
+        trigger_motor->geared_down_pos_set = trigger_motor->geared_down_pos_raw + TRIGGER_90_DEGS;
+        trigger_motor->cmd_time = xTaskGetTickCount();
+        trigger_motor->move_flag = 1;
+    }
+
+    //Check for whether target is reached
+    if (trigger_motor->geared_down_pos_set - trigger_motor->geared_down_pos_raw > TRIGGER_REACHED_POS_RANGE)
```
Comment (jamesrosstwo) [inline user/TASK/shoot_task/shoot_task.c]: If we're getting the difference and checking if it's within a range, should this be an absolute value? Geared_down_pos_set might also need to be refactored for clarity

File: user/TASK/shoot_task/shoot_task.h [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+#define HALF_TRIGGER_RATIO 18
+#define FULL_TRIGGER_RATIO 36
+#define TRIGGER_REACHED_POS_RANGE 500
+#define TRIGGER_SPEED 1000
+#define TRIGGER_OFF 0
+
+
+typedef struct
+{
+	const motor_data_t *shoot_motor_raw;
```
Comment (jamesrosstwo) [inline user/TASK/shoot_task/shoot_task.h]: Try to maintain consistency with spaces vs. tabs

**Pull request closed without merging**